packages/perseus-editor/src/components/locked-vector-settings.tsx

```diff
@@ -0,0 +1,170 @@
+    AccordionProps & {
+        /**
+         * Called when the delete button is pressed.
+         */
+        onRemove: () => void;
+        /**
+         * Called when the props (points, color, etc.) are updated.
+         */
+        onChangeProps: (newProps: Partial<LockedFigure>) => void;
+    };
+
+const LockedVectorSettings = (props: Props) => {
+    const [tailCoordExpanded, setTailCoordExpanded] = useState(false);
+    const [tipCoordExpanded, setTipCoordExpanded] = useState(false);
+    const {points, color: lineColor, onChangeProps, onRemove} = props;
+    const [tail, tip] = points;
+    const lineLabel = `Vector (${tail[0]}, ${tail[1]}), (${tip[0]}, ${tip[1]})`;
+
+    // Check if the line has length 0.
+    const isInvalid = kvector.equal(tail, tip);
+
+    function handleChangePoint(newCoord: Coord | undefined, index: 0 | 1) {
+        if (typeof newCoord !== "undefined") {
+            const newPoints = [...points] as [Coord, Coord];
+            newPoints[index] = [...newCoord];
+            onChangeProps({
+                points: newPoints,
+            });
+        }
+    }
+
+    function handleColorChange(newColor: LockedFigureColor) {
+        onChangeProps({
+            color: newColor,
+        });
+    }
+
+    return (
+        <LockedFigureSettingsAccordion
+            expanded={props.expanded}
+            onToggle={props.onToggle}
+            header={
+                <View style={styles.row}>
+                    <LabelLarge>{lineLabel}</LabelLarge>
+                    <Strut size={spacing.xSmall_8} />
+                    <LineSwatch color={lineColor} lineStyle="solid" />
+                </View>
+            }
+        >
+            <View style={[styles.row, styles.spaceUnder]}>
+                {/* Line color settings */}
+                <ColorSelect
+                    selectedValue={lineColor}
+                    onChange={handleColorChange}
+                />
+            </View>
+
+            {/* Zero length error message */}
+            {isInvalid && (
+                <LabelMedium style={styles.errorText}>
+                    {lengthErrorMessage}
+                </LabelMedium>
+            )}
+
+            <LockedFigureSettingsAccordion
+                expanded={tailCoordExpanded}
+                onToggle={() => setTailCoordExpanded(!tailCoordExpanded)}
+                containerStyle={styles.container}
+                panelStyle={styles.accordionPanel}
+                header={
+                    // Summary: Point, coords, color (filled/open)
+                    <View style={styles.row}>
+                        <LabelLarge>{`Tail (${tail[0]}, ${tail[1]})`}</LabelLarge>
+                    </View>
+                }
+            >
+                <CoordinatePairInput
+                    coord={tail}
+                    error={isInvalid}
+                    onChangeProps={(newProps) => {
+                        handleChangePoint(newProps.coord, 0);
```

Comment:
Oh also: I think Matthew commented on my circle comment about this being unintuitive. I have an update coming for `CoordinatePairInput`. I wonder if it'll help.

https://github.com/Khan/perseus/pull/1326/commits/034dda463f764c448e17c6be1e29b1377c165605